Rework mgmt as team preparation (#57)
data-lessons/instructor-training: _episodes/18-management.md

---
title: "Working With Your Team"
block: "Building Teaching Skill"
teaching: 30
exercises: 45
questions:
- "What are the challenges of managing a heterogeneous classroom?"
- "What should we do if there is a Code of Conduct violation?"
- "What does it mean to be a co-Instructor?"
- "How does an instructional team go about setting up for a workshop?
objectives:
- "Identify potential challenges of teaching learners with very different backgrounds and skill levels."

> - "Locate resources to direct your response if someone at your workshop violates the Code of Conduct."

Author: The problem is that we don't provide enough training to reasonably expect that trainees will know what to do. I changed this because I was actively lowering expectations for what we can accomplish here, and deferring actual training on CoC response to modules that do address that. 
I'd be happy to consider another alternative, but I think the original objective is not one we can meet.

- "Identify workshop roles and responsibilities for your team."
- "Use The Carpentries workshop website template instructions to start creating a website."
keypoints:
- "Team work takes work, but allows you to share the load and build connections."
- "Working with a broad range of learners can be challenging, but there are many ways to keep a classroom happy and motivated."

> - "The instructional team decides how to respond to Code-of-Conduct incidents during a workshop; all violations should be reported to The Carpentries for follow-up."

Reviewer: ```suggestion
- "The instructional team decides how to respond to Code-of-Conduct incidents during a workshop; all violations should be reported to The Carpentries Code of Conduct committee for follow-up."
```

Reviewer: ?

Author: Did I get this wrong?

you should also have a crew of Helpers to answer questions individually, elevate common problems for general discussion, and make sure the pacing 
works for everyone.

The people who are planning and executing your workshop with you can really lighten the load! They are also there to
help if you find yourself in a sticky situation. However, from logistics planning to delicate problem solving, a little planning goes a long
way, and all co-Instructors should be in agreement about how they will support each other in engaging challenges. 
In this episode, we will discuss examples, strategies, and resources for handling challenges you may face in the classroom. We will
also take some time to discuss planning logistics, and (time permitting) everyone will have an opportunity to try out
our instructions for setting up a workshop website.

## The Instructional Team
A typical Carpentries workshop includes 3 roles; sometimes individuals may choose to play more than one role. Each workshop should have:
- A **Host** who organizes the workshop logistics
- Two or more **Instructors** who plan and execute workshop instruction

> - **Helpers** who provide learner support during the workshop  

Reviewer: ```suggestion
- **Helpers** who support learners during the workshop  
```

When Instructors self-organise a workshop, they sometimes find themselves playing the role of Host as well as Instructor.
We recommend avoiding this if possible, because hosting responsibilities will subtract from the time you have
to prepare and teach. 

If you cannot find someone else to bear full responsibility for the Host role, consider recruiting help for specific tasks. This may include 
involving helpers in advance of the workshop, or exploring institutional support for event logistics like registration, snack delivery, 
helper coordination, and emergency contact.

### Helpers

[Helpers][helpers] are usually recruited from a local community by the Host of a workshop. Helpers may be
involved prior to the day of the workshop, but often they simply show up. In most cases, helpers are expected to 
attend for the full workshop, but in some communities they may come and go during different segments.

Helpers can assist in a variety of ways. They may:
- help learners with setup and installation
- answer questions during exercises
- monitor the room to spot people who may need help (indicated by a sticky note or otherwise)

> - keep an eye on the shared notes and either answer questions there or remind the instructor to do so during breaks

Reviewer: ```suggestion
- keep an eye on the shared notes and either answer questions there or remind the Instructor to do so during breaks
```

> Some instructors use [Virtual Private Servers][vps] (VPS) over [Secure Shell][ssh]
> (SSH) or web browser pages instead. This solves the installation
> issues, but makes us dependent on host institutions' WiFi (which can be
> unreliable), and has the issues mentioned above with things
> like keyboard shortcuts.
{: .callout}

#### Sticky Notes
 At in-person workshops, two different colors of paper sticky notes are given to each learner. These act as 'flags' that may be placed 
 on a laptop screen to indicate when a learner needs help or when they have completed a task and are ready to move on. 
 This is better than having people raise their hands because:  
*  it is more discreet (which means they are more likely to actually do it),
*  they can keep working while their flag is raised, and
*  the instructor can quickly see from the front of the room what state the class is in.  

>  **Active and frequent requests** are most effective in getting learners to actually use these well (e.g. "lease put up a blue sticky note if

Reviewer: ```suggestion
 **Active and frequent requests** are most effective in getting learners to actually use these well (e.g. "please put up a blue sticky note if
```

 you completed the last step, and a yellow sticky note if you need help.").  

>  In online workshops, Zoom has a green "✓" and red "x" that may be used in place of sticky notes.

Reviewer: ```suggestion
 In online workshops, Zoom has a green "✓" and red "x" under "reactions" that may be used in place of sticky notes.
```

> because of their common symbolism for "stop" and "go." However, we have learned 
> and grown! We strongly encourage you to get sticky notes in colors other than red and green (e.g. blue and yellow), 
> because learners, helpers, and Instructors with colorblindness may have difficulty
> distinguishing red from green. 
> Some Instructors have also had great fun with differently-shaped sticky notes (stars and squares, 
> thumbs up that can be turned into thumbs down stickies etc.).
{: .callout}

 #### Formative Assessment
 Calling for sticky notes is one kind of formative assessment! But it does not give you detailed information about how learners
 are progressing. 
 
#### Breaks (ideally with snacks)
 Two-day workshops often fit best in busy schedules, allowing an intensive burst of dedicated time and energy. However, "intensive" is 
 a key word -- two day workshops are intense! In addition to minding your pacing, it is important to schedule breaks, and to take them 
 on time. In addition to allowing vital rest and social time, this will also allow learners to attend to work and personal matters at predictable times.  

>  At in-person workshops, coffee and snacks to suit a range of dietary needs are highly recommended. This may mean charging a 

Reviewer: ```suggestion
 At in-person workshops, beverages and snacks to suit a range of dietary needs are highly recommended. This may mean charging a 
```

- **Teach and assist:** Teacher A teaches while Teacher B moves around
  the classroom to help learners.

"Teach and assist" is most common, in part because people are accustomed to teaching alone! However, 
"Team teaching" can be a fun way to lead together as well. Team teaching can be particularly helpful when two 
Instructors have very different strengths, creating a more balanced experience overall. This may also be worth 

> considering for self-organised workshops in a co-Instructor has not yet been certified.

Reviewer: ```suggestion
considering for self-organised workshops if a co-Instructor has not yet been certified.
```

away feeling satisfied that their time was well spent.

> ## What Are the Challenges?
> What are some of the challenges you might expect when teaching learners with a broad range of expertise? Add your thoughts in the Etherpad.
>
> This discussion should take about 5 minutes.
{: .challenge}

Here are some strategies Carpentries
Instructors have generated to deal with this issue:

*   Be sure that workshop **advertisements communicate its level clearly**
    by describing not only the topics (e.g. plotting in R), but
    also the specific skills and tasks (i.e. the learning objectives).
    If advanced learners do sign up anyway (and they may!), this will be an informed choice.
*   Consider creating **"beginner" and "advanced" options** for exercises.
    Learners who finish the "beginner" exercise can then challenge themselves with a next step.
*   During class, **encourage learners to help others** nearby when they feel comfortable doing so.
    Teaching is a great way to level up learning (one of many reasons why Carpentries Instructors choose to teach!).
    Note that you will need an icebreaker before most learners will take this advice.

> *   **Do not let advanced learners carry the conversation** during class, no matter how interesting it may be. This can alienate novices

Reviewer: ```suggestion
*   **Do not let advanced learners take over the conversation** during class, no matter how interesting it may be. This can alienate novices
```

    and consumes precious time. Advanced questions and comments can be politely reserved for breaks, addressed
    individually by helpers or the co-Instructor, or typed in the Etherpad for present or future follow-up.
*   The helpers and non-teaching co-Instructor(s)
    should **be vigilant for learners who are falling behind**
    and intervene early
    so that they do not become frustrated and give up. Frequent reminders
    to put up sticky notes for "ok" or "I'm stuck" will support your 
    team in targeting problems in time.

> No class can possibly meet all individual training needs.

Reviewer: I absolutely love this whole paragraph!

Dealing effectively with different skill levels does take some planning. However, with
appropriate advertising and team cohesion on priorities and strategies, your workshop
can be a worthwhile experience for everyone.

## Sticky Situations 2: Code of Conduct Violations

As discussed in the [Motivation episode]({{ page.root }}{% link _episodes/08-motivation.md %}),
the Carpentries uses a Code of Conduct as a tool to create a respectful and
inclusive learning environment. This works well! But Codes of Conduct do not hold up if they
are not enforced. Rarely, your instructional team
may encounter a Code of Conduct violation. In this section, we will discuss resources available to
help you prepare and respond appropriately in the event that a violation occurs.

All Carpentries community members should feel empowered to enforce the Code of Conduct (CoC). However, you may not
feel comfortable taking action alone, and you may wish to be informed when a team-member takes action in your workshop. 
For these reasons and more we encourage you to **discuss Code of Conduct enforcement with your instructional team in advance**. 

> ## Know Your Resources

> > 1) Take 5 minutes to read through the Code of Conduct Incident Response Guidelines: https://docs.carpentries.org/topic_folders/policies/incident-response.html

Reviewer: ```suggestion
> 1) Take 5 minutes to read through the [Code of Conduct Incident Response Guidelines](https://docs.carpentries.org/topic_folders/policies/incident-response.html)
```

Author: In the exercises, I am leaving links in the text intentionally because this facilitates transfer to the Etherpad.

> > 2) Discuss what you have read in small groups. As questions arise, you may wish to refer to our complete Code of Conduct section in The Carpentries Handbook: https://docs.carpentries.org/topic_folders/policies/index_coc.html  

Reviewer: ```suggestion
> 2) Discuss what you have read in small groups. As questions arise, you may wish to refer to our complete [Code of Conduct section in The Carpentries Handbook](https://docs.carpentries.org/topic_folders/policies/index_coc.html)  
```

> * What would you want to discuss or have your instructional team agree upon in advance of your workshop? 
> * What questions do you have about CoC enforcement? 

> > 3) Write some notes in the etherpad. 

Reviewer: ```suggestion
> 3) Write some notes in the Etherpad. 
```

the person you just taught with.

### Setting up a Workshop Website

In addition to planning how the workshop will go, someone on your instructional team 
will be charged with creating a workshop website. This website helps your learners to
know what to expect. It also helps The Carpentries to keep track of your workshop and 
to credit everyone on your team.

The Carpentries maintains a template repository for these websites with step-by-step 
instructions for setting them up.

> ## Practice With The Carpentries Infrastructure
> 
> For this activity, your Trainer will put you in groups, but you may choose whether to work together or independently. 
> If you work independently, you can still use your group as a resource to ask questions as they emerge. 
>
> Go to the [workshop template repository]({{ site.workshop_repo }}). 
> 

> > * If you have a GitHub account (or don't mind quickly creating one) and are comfortable doing so, 

Reviewer: ```suggestion
> * If you have a GitHub account (or don't mind creating one) and are comfortable doing so, 
```

> 
> * Alternatively, have a look at the video tutorial linked on the instructions page. With any time
> remaining, have a look at the websites for [upcoming Carpentries workshops][https://carpentries.org/upcoming_workshops/] on our website.
> 
> * Add your questions and thoughts on this process to the Etherpad. If you created a workshop website,
> add the link there as well.
>
> This exercise should take about 15 minutes. 
>
> Note: Sometimes web browsers will cache the workshop webpage, so when
> you make changes in GitHub, they do not show up on the workshop webpage
> immediately.  Two ways to avoid this are to use a "private" or
> "incognito" mode in your web browser or by following these
> [instructions to bypass your browser cache](https://en.wikipedia.org/wiki/Wikipedia:Bypass_your_cache).
{: .challenge}

You may also wish to create an Etherpad for your workshop
(or other Carpentries-related event).
When you point a browser to a url at
`https://pad.carpentries.org/workshop-slug`
(e.g., `https://pad.carpentries.org/2022-02-22-data-skills`
for the example above)
you will either arrive at an existing Etherpad by that name or a brand new

> etherpad with basic Carpentries information at the top.

Reviewer: ```suggestion
Etherpad with basic Carpentries information at the top.
```
